webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -620,60 +620,100 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
      * @return the calculated font size that makes the text fit
      */
     public static float fitText(Font font, String text, Rectangle rect, float maxFontSize, int runDirection, boolean newlinesAllowed) {
-        try {
-            ColumnText ct = null;
-            int status = 0;
-            if (maxFontSize <= 0) {
-                int cr = 0;
-                int lf = 0;
-                char[] t = text.toCharArray();
-                for (int k = 0; k < t.length; ++k) {
-                    if (t[k] == '\n') {
-                        ++lf;
-                    } else if (t[k] == '\r') {
-                        ++cr;
-                    }
-                }
-                int minLines = Math.max(cr, lf) + 1;
-                maxFontSize = Math.abs(rect.getHeight()) / minLines - 0.001f;
-            }
-            font.setSize(maxFontSize);
-            Phrase ph = new Phrase(text, font);
-            ct = new ColumnText(null);
-            ct.setSimpleColumn(ph, rect.getLeft(), rect.getBottom(), rect.getRight(), rect.getTop(), maxFontSize, Element.ALIGN_LEFT);
-            ct.setRunDirection(runDirection);
-            status = ct.go(true);
-            if ((status & ColumnText.NO_MORE_TEXT) != 0 && (newlinesAllowed || ct.getLinesWritten() <= 1)) {
-                return maxFontSize;
-            }
+
+        //first verify if there's a line break ("\n") -> megaminx
+        //if so, calculate the font based only on the width
+        int lineBreak = text.indexOf("\n");
+
+        if(lineBreak > 0) {
+            //the char width is fontSize * 0.6 -> this was found with getWidthPoint()
+            float size = rect.getWidth() / (lineBreak * 0.6f);
+
+            //the calculated size is a little too big, so we reduce it to 95%
+            return (size * 0.95f);
```

Comment:
Rather than multiplying by 0.95 here, lets increase the margins in the code that calls `fitText`.